lightning/src/util/events.rs

```diff
@@ -254,6 +255,14 @@ pub enum Event {
 		channel_id: [u8; 32],
 		/// The reason the channel was closed.
 		reason: ClosureReason
+	},
+	/// Used to indicate to the user that they can abandon the funding transaction and recycle the
+	/// inputs for another purpose.
+	DiscardFunding {
+		/// The channel_id of the channel which has been closed.
+		channel_id: [u8; 32],
+		/// The full transaction received from the user
+		transaction: Option<Transaction>
```

Comment:
I'm not sure why this needs to be an option - `is_funding_pending` is gated on `funding_transaction.is_some()`, so we could combine it and `funding_transaction` into one function that checks the `channel_state` is as expected, and then returns the transaction if there is one, otherwise returns None.